src/main/java/com/facebook/openwifirrm/modules/Modeler.java

```diff
@@ -93,8 +95,9 @@ public static class DataModel {
 		public Map<String, List<List<WifiScanEntry>>> latestWifiScans =
 			new ConcurrentHashMap<>();
 
-		/** List of latest state per device. */
-		public Map<String, State> latestState = new ConcurrentHashMap<>();
+		/** List of latest states per device. */
+		public Map<String, List<State>> latestStates =
```

Comment:
this needs to be thread safe. we need to use thread safe structures and/or proper locking. As is there will be race conditions and weird behaviors (i don't know the full behavior of java but i imagine it could crash given certain code paths).